doc/user/content/ingest-data/sizing.md

```diff
@@ -0,0 +1,357 @@
+WHERE c.name = 'src_cluster'
+GROUP BY s.id, s.name
+```
+
+To determine how many Kafka messages (and how many bytes worth) Materialize has read, `messages_received` and `bytes_received` respectively. However,
+its currently not possible to easily determine whether or not _hydration_ is finished. You can wait for the memory usage of your cluster to reduce
+and level-out using the guidance in [Inspecting resource utilization](#inspecting-resource-utilization).
+
+{{< note >}}
+In the future there may be additional system tables that simplify determining if hydration is finished.
+{{</ note >}}
+
+If you see OOM's happening, you may need to increase your `hydration_size`, and use:
+```sql
+ALTER CLUSTER src_cluster SET SIZE = <new_hydration_size>
+```
+or, wait for some sources to hydrate before `CREATE`-ing others.
+
+
+### Shrinking and testing rehydration
+After _hydration_ is finished, you may run:
+```sql
+ALTER CLUSTER src_cluster SET SIZE = <steady_state_size>
+```
+
+to shrink your cluster to its steady state size. This also ensures that your cluster can _rehydrate_ properly,
+which typically uses more CPU and memory than _steady_state_ (even for disk-attached replicas). You can
+use [Inspecting resource utilization](#inspecting-resource-utilization) to see the resource utilization of
+_rehydration_, and use:
+```sql
+SELECT
+  bool_and(mss.rehydration_finished IS NOT NULL)
+FROM mz_internal.mz_source_statistics mss
+  JOIN mz_sources s ON s.id = mss.id
+  JOIN mz_clusters c ON c.id = s.cluster_id
+WHERE c.name = 'src_cluster';
+```
+to determine when the process is finished.
+
+
+{{< note >}}
+If your `hydration_size` and `steady_state_size` are the same, we still recommend you test _rehydration_, by running the following:
+```sql
+ALTER CLUSTER src_cluster SET (REPLICATION FACTOR = 0);
+ALTER CLUSTER src_cluster SET (REPLICATION FACTOR = 1);
+```
+{{</ note >}}
+
+### Inspecting your working set
+You can run:
+```sql
+SELECT
+  SUM(mss.envelope_state_count) ingested_working_set_total_keys
+  SUM(mss.envelope_state_bytes) / 1024 / 1024 / 1024 as ingested_working_set_gib
+FROM mz_internal.mz_source_statistics mss
+  JOIN mz_sources s ON s.id = mss.id
+  JOIN mz_clusters c ON c.id = s.cluster_id
+WHERE c.name = '<your cluster name>';
+```
+
+to determine the total number of keys and size of the _working sets_ of your upsert-like sources. If you want it broken
+out by source, run this instead:
+```sql
+SELECT
+  s.id,
+  s.name,
+  SUM(mss.envelope_state_count) ingested_working_set_total_keys
+  SUM(mss.envelope_state_bytes) / 1024 / 1024 / 1024 as ingested_working_set_gib
+FROM mz_internal.mz_source_statistics mss
+  JOIN mz_sources s ON s.id = mss.id
+  JOIN mz_clusters c ON c.id = s.cluster_id
+WHERE c.name = '<your cluster name>'
+GROUP BY s.id, s.name;
+```
+
+
+## Append-only sources
+Postgres sources and non-Upsert/Debezium Kafka sources are _append-only_, which means they _rehydrate_ effectively immediately.
```

Comment:
Would it be better to say that there's no rehydration since there's no state to keep? Because the `rehydration_latency_ms` is not going to be populated for these.